src/ImageSharp/Formats/Jpeg/Components/Block8x8F.cs

```diff
@@ -27,29 +27,64 @@ internal partial struct Block8x8F : IEquatable<Block8x8F>
         public const int Size = 64;
 
 #pragma warning disable SA1600 // ElementsMustBeDocumented
+        [FieldOffset(0)]
         public Vector4 V0L;
+        [FieldOffset(16)]
         public Vector4 V0R;
 
+        [FieldOffset(32)]
         public Vector4 V1L;
+        [FieldOffset(48)]
         public Vector4 V1R;
 
+        [FieldOffset(64)]
         public Vector4 V2L;
+        [FieldOffset(80)]
         public Vector4 V2R;
 
+        [FieldOffset(96)]
         public Vector4 V3L;
+        [FieldOffset(112)]
         public Vector4 V3R;
 
+        [FieldOffset(128)]
         public Vector4 V4L;
+        [FieldOffset(144)]
         public Vector4 V4R;
 
+        [FieldOffset(160)]
         public Vector4 V5L;
+        [FieldOffset(176)]
         public Vector4 V5R;
 
+        [FieldOffset(192)]
         public Vector4 V6L;
+        [FieldOffset(208)]
         public Vector4 V6R;
 
+        [FieldOffset(224)]
         public Vector4 V7L;
+        [FieldOffset(240)]
         public Vector4 V7R;
+
+#if SUPPORTS_RUNTIME_INTRINSICS
+        [FieldOffset(0)]
+        public Vector256<float> V0;
+        [FieldOffset(32)]
+        public Vector256<float> V1;
+        [FieldOffset(64)]
+        public Vector256<float> V2;
+        [FieldOffset(96)]
+        public Vector256<float> V3;
+        [FieldOffset(128)]
+        public Vector256<float> V4;
+        [FieldOffset(160)]
+        public Vector256<float> V5;
+        [FieldOffset(192)]
+        public Vector256<float> V6;
+        [FieldOffset(224)]
+        public Vector256<float> V7;
+#endif
```

Comment:
@saucecontrol are you aware of any codegen bugs or perf issues with `[FieldOffset]` + intrinsics?

Reply:
There shouldn't be any, either way a lot of avx code already assumes that left/right vectors are aligned this way:
```csharp
Avx.Multiply(Unsafe.As<Vector4, Vector256<float>>(ref this.V0L), valueVec)
```
If there are any issues with that, all `Unsafe.As` casts in avx should be affected.

Reply:
Just for clarity: this is not about layout, your explicit layout is equivalent to the implicit sequential layout before the change.

I expect the runtime having different implementations for codegening `Unsafe.As` and `[FieldOffset]`. Normally they should be equivalent, but we've seen a couple of weird bugs in the past around similar areas.

While I trust `Unsafe.As`, seeing `[FieldOffset]` together with intrinsics kinda frightens me :) Let's see if @saucecontrol can react.

Reply:
I see, never thought this could cause any troubles, interested to hear his opinion on this :)

Reply:
I'm not aware of any issues, but I'd be happy to pull this down and give the codegen a look.  Will try to get to that this afternoon.

Reply:
@saucecontrol thanks!

Reply:
Codegen around the ExplicitLayout struct looks good.  The codegen around the vector constants could be a bit better, for example, from the FDCT:

```C#
d.V2 = SimdUtils.HwIntrinsics.MultiplyAdd(Avx.Multiply(c2, C_V_0_5411), c3, C_V_1_3065);
d.V6 = SimdUtils.HwIntrinsics.MultiplySubtract(c3, C_V_0_5411, Avx.Multiply(c2, C_V_1_3065));
```
(assuming you get that `MultiplySubtract` helper in 😸)
compiles to:

```asm
mov         rax,16E6ABBA590h                 ' load C_V_0_5411 offset
mov         rax,qword ptr [rax]              ' load C_V_0_5411 address
vmulps      ymm9,ymm4,ymmword ptr [rax+8]  
mov         rax,16E6ABBA598h                 ' load C_V_1_3065 offset
mov         rax,qword ptr [rax]              ' load C_V_1_3065 address
vfmadd231ps ymm9,ymm2,ymmword ptr [rax+8]  
vmovupd     ymmword ptr [rdx+40h],ymm9  
mov         rax,16E6ABBA590h                 ' C_V_0_5411 again
mov         rax,qword ptr [rax]              ' "
mov         rcx,16E6ABBA598h                 ' C_V_1_3065 again  
mov         rcx,qword ptr [rcx]              ' "
vmulps      ymm4,ymm4,ymmword ptr [rcx+8]  
vfmsub132ps ymm2,ymm4,ymmword ptr [rax+8]
vmovupd     ymmword ptr [rdx+0C0h],ymm2  
```

You could try loading them into temp locals for re-use, but I'm not sure if you'll run out of SIMD registers doing that.  I only see up to ymm10 being used, so you should have a few to play with.